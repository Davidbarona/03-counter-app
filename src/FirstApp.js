import React from "react";
import PropTypes from 'prop-types';

const FirstApp = ({ greeting,subtitle}) => {
  // const greeting = 'hello Bro'
  // const object = {
  //     name:'dave',
  //     age:29
  // }
//   console.log(props);

    if(!greeting){
        throw new Error ('you need put the greeting')
    }

  return (
    <>
      <h1>{greeting}</h1>
      {/* <pre>{JSON.stringify(object,null,3)}</pre> */}
   
      <p>{subtitle}</p>
    </>
  );


};

FirstApp.propTypes={
    greeting:PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    subtitle:"I'm subtitle"
}

export default FirstApp;
