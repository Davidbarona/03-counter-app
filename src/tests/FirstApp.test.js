import React from 'react'
import { render } from "@testing-library/react"
import FirstApp from "../FirstApp"
import {shallow} from 'enzyme'
import '@testing-library/jest-dom'

describe('testing on First App',()=>{
    
    // Don't will function because tobeInthe document is from @testing-library/jest-dom/extend-expect and not it's turn on
    // test("should of show Hi,I'm Goku", () => {
        
    //     const greeting = "Hi,I'm Goku"
    //     const {getByText}= render(<FirstApp greeting={greeting}/>)

    //     expect(getByText(greeting)).toBeInTheDocument()
    // })

    test('should render <FirstApp /> correctly', () => {
        
        const greeting = "Hi,I'm Goku"
        const wrapper = shallow(<FirstApp greeting={greeting}/>)
        expect(wrapper).toMatchSnapshot();
    });

    test('should render the subtitle send by props', () => {
        const greeting = "Hi,I'm Goku"
        const subtitle = "Hi,I'm a subtitle"
        const wrapper = shallow(
        <FirstApp
             greeting={greeting}
             subtitle={subtitle}
             />)

        const paragraphText = wrapper.find('p').text()
        
        expect(paragraphText).toBe(subtitle)
        
    });
    
    
    
})