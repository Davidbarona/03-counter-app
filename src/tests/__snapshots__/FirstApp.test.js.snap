// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing on First App should render <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hi,I'm Goku
  </h1>
  <p>
    I'm subtitle
  </p>
</Fragment>
`;
